actions.js

```diff
@@ -0,0 +1,56 @@
+
+/* eslint-disable import/prefer-default-export */
+import { Toast } from 'native-base';
+import CONFIG from './config.js';
+
+const API_URL = CONFIG.ENV === 'PROD' ? CONFIG.API_URL : CONFIG.TEST_API_URL;
+
+export function constructContactItemsWithToken(token) {
+  return async (dispatch) => {
+    try {
+      dispatch({
+        type: 'SET_USER_DATA',
+        data: {
+          refreshingContacts: true
+        }
+      });
+      const response = await fetch(`${API_URL}/contacts/${token._id}?token=${token.api_key}`);
+      if (!response.ok) {
+        throw new Error(`Failed with status code: ${response.status}`);
+      }
+      const contactData = (await response.json()).contacts.map(contact => ({
+        name: `${contact.person.fname} ${contact.person.lname}`,
+        school: contact.school.name,
+        grade: contact.school.grade,
+        id: contact._id,
+        facebook_id: contact.facebook_id,
+        fullContact: contact
+      }));
+      if (contactData.length === 0) {
+        Toast.show({
```

Comment:
I'm not too familiar with redux and thunk, but is it a general model to have the actions produce error messages that are surfaced to the user from the action?

Reply:
we can either send the errors back and let the view handle them or address this globally. As a first step, I like to address them in a generic way so that we don't have to rewrite the way they are addressed in each view. For more fine grained control, we would need to transfer those errors back and handle them multiple times according to the view

Reply:
Maybe it makes sense to have an error field on the state that gets updated, instead of the actions knowing about Toast and how to display errors. 

I'd prefer that the actions be as decoupled from the mobile app logic like error surfacing as possible.

Reply:
makes sense, changed